src/daft-plan/src/optimization/rules/push_down_filter.rs

```diff
@@ -509,33 +482,26 @@ mod tests {
     fn filter_commutes_with_join_both_sides() -> DaftResult<()> {
         let source1: LogicalPlan = dummy_scan_node(vec![
             Field::new("a", DataType::Int64),
-            Field::new("b", DataType::Utf8),
+            Field::new("b", DataType::Int64),
```

Comment:
Hmm why the change of this dtype to an `Int64` and the filter predicate to be on column `"b"`. instead of column `"c"`?

The intention for this test was to have a join on a string column (e.g. some IDs) and to have a filter on a non-join-key column (column `"c"`) that's shared by both sides of the join and is pushed down into each side of the join.

Reply:
Ah duh, I forgot about the typical renaming behavior for duplicate column names! Yeah I was hoping to approximate the latter, but I think having the filter be on the join key column is simpler and preferable for this test, so I think we should keep it as is!